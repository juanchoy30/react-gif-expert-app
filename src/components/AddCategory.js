import React, { useState } from 'react'

const AddCategory = () => {

    const [inputValue, setInputValue] = useState('Hello World');

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('submit done')
    };

    return (
        <form onSubmit={ handleSubmit }>
            <input
                type="text"
                value={ inputValue }
                onChange={ handleInputChange } 
            />
        </form>
    );
}

export default AddCategory;