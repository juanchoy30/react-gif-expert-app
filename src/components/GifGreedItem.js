import React from 'react'

export const GifGreedItem = ({ id, title, url }) => {

    return (
        <div className="card">
            <img src={ url } alt={ title } />
            <p> { title } </p>
        </div>
    )
}
