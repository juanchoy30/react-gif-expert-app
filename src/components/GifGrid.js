import React, { useState, useEffect } from 'react'
import { getGif } from '../helpers/getGifs';
import { GifGreedItem } from './GifGreedItem';

const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect( () => {
        getGif( category )
            .then( setImages );  // it is the same as .then( imgs => setImages( imgs ));
    }, [ category ]); // [] executes the function just once, if not the function could execute in an infinite cycle

    return (
        <>
            <h3>{ category }</h3>
            <div className="card-grid">
                {
                    images.map( img => (
                        <GifGreedItem 
                            key={ img.id }
                            { ...img }
                        />
                    ))
                }
            </div>
        </>

    )
}

export default GifGrid;