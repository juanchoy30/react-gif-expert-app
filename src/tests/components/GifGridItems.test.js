import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('<GifGridItem /> tests', () => {

    const title = 'A title';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( 
        <GifGridItem
            title={ title }
            url ={ url }
        /> 
    );
    
    test('should show the component correctly', () => {

        expect( wrapper ).toMatchSnapshot();

    })

    test('should contain a paragraph with title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    })

    test('should contain the imgÂ´s url and alt equal to those of the props', () => {
        
        const img = wrapper.find('img');
        //console.log( img.props('src') );
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    })

    test('should have animate__fadeIn class', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');

        console.log( div.prop('className') );

        expect( className.includes('animate__fadeIn') ).toBe( true );
        
    })
    
    
})
