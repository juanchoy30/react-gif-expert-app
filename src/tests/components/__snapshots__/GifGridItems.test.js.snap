// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> tests should show the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    A title
     
  </p>
</div>
`;
